File: two_pointers/minimizeAbsoluteDiff.cpp
#include<iostream>
#include<vector>
#include<cmath> 
#include<algorithm>
#include<set>
#include<string>
#include<map>
#include<cstring>
#include<stack>
#include<queue>

using namespace std;

#define ll long long

const int INF = 1e9+7;
const int mod = 1e9+7;
const int MAX = 1e5+5;

int solve(vector<int> &A, vector<int> &B, vector<int> &C){
	int diff = MAX;
	int i = 0, j = 0, k = 0;

	while(i < A.size() && j < B.size() && k < C.size()){
		int low = min(min(A[i], B[j]), C[k]);
		int high = max(max(A[i], B[j]), C[k]);

		diff = min(diff, high-low);

		// advance index of the array with minimum value
		if(A[i] == low)
			i++;
		else if(B[j] == low)
			j++;
		else
			k++;
	}

	return diff;
}

int main(){
	//freopen("tree.in","r",stdin);
    //freopen("tree.out","w",stdout);
    //freopen("input.txt", "r", stdin);
    //freopen("out.txt", "w", stdout);
    ios_base::sync_with_stdio(0);
    cin.tie(0);

    vector<int> A{3, 6, 8, 10, 15};
    vector<int> B{1, 5, 12};
    vector<int> C{4, 8, 15, 16};

    cout << solve(A, B, C) << endl;
}

File: two_pointers/minimizeAbsoluteDiff.py

def solve(A, B, C):
	diff = (10**9)+5
	i, j, k = 0, 0, 0
	while(i < len(A) and j < len(B) and k < len(C)):
		low = min(min(A[i], B[j]), C[k])
		high = max(max(A[i], B[j]), C[k])

		diff = min(diff, high-low)

		if A[i] == low:
			i += 1
		elif B[j] == low:
			j += 1
		else:	
			k += 1
	return diff				



A = [int(i) for i in input().split()] #[3, 6, 8, 10, 15]
B = [int(i) for i in input().split()] #[1, 5, 12]
C = [int(i) for i in input().split()] #[4, 8, 15, 16]
print(solve(A, B, C))